server/grpcsvr/bmc/machine.go

```diff
@@ -19,167 +19,152 @@ type MachineAction struct {
 	StatusMessages    chan string
 }
 
+type power interface {
+	connection() repository.Error
+	close()
+	on() (string, repository.Error)
+	off() (string, repository.Error)
+	status() (string, repository.Error)
+	reset() (string, repository.Error)
+	hardoff() (string, repository.Error)
+	cycle() (string, repository.Error)
+}
+
 // BootDevice functionality for machines
-func (p MachineAction) BootDevice() (string, repository.Error) {
-	var result string
-	errMsg := repository.Error{
-		Code:    0,
-		Message: "",
-		Details: nil,
-	}
-	l := p.Log.GetContextLogger(p.Ctx)
-	l.V(0).Info("not implemented")
-	msg := "power OFF not implemented"
-	l.V(1).Info(msg)
+func (m MachineAction) BootDevice() (result string, errMsg repository.Error) {
+	l := m.Log.GetContextLogger(m.Ctx)
+	msg := "setting Boot Device not implemented yet"
+	l.V(0).Info(msg)
```

Comment:
hmm I think it would make sense to embed into `machinAction` an api for logging that does this boiler plate stuff for us, not in this PR obviously, but thoughts?

Reply:
ah yes, that's a good idea. will put it on the list to add. would be a pretty straightforward change too, I believe.

Reply:
nice :+1: